src/features/expenses/AddExpense/MainInfo.tsx

```diff
@@ -0,0 +1,142 @@
+import { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+
+import { Input, Select, TextUnderline } from "components";
+import { Column, PageWrapper, Row } from "components/Containers";
+import Header from "components/Header";
+import { ReactComponent as BackIconSVG } from 'assets/images/back-icon.svg';
+import { ReactComponent as ArrowRightSVG } from 'assets/images/arrow-right.svg';
+import { BackgroundFiller, ContentWrapper, CurrencyText, HalfCircleBackground, MainInfoText, MainInfoWrapper, NextStepButtonWrapper } from "./MainInfo.styled";
+import BottomNavigation from "components/BottomNavigation";
+import currencies from "constants/currencies";
+import { ICreateTransaction } from "../types";
+import { newTransactionSelector } from "../slice";
+import { CreateRouteString, ExpenseRouteMode, composeExpenseRoute } from "constants/routes";
+import { saveTransaction } from "../slice";
+
+interface ILocationState {
+  proceed?: boolean;
+}
+
+const MainInfo = () => {
+  const { t } = useTranslation('expenses');
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const locationState = location.state as ILocationState;
+
+  const {groupId: groupIdOptional} = useParams();
+  const groupId = groupIdOptional ?? '';
+
+  const transaction = useSelector(newTransactionSelector);
+
+  const [form, setForm] = useState<ICreateTransaction>({
+    name: '',
+    paymentDateUtc: new Date().toISOString().split('T')[0],
+    currencyCode: currencies[0].value,
+    groupId: groupId,
+    payerDetails: [],
+    partialsAssignments: [],
+  });
+
+  const handleBackAction = useCallback(() => {
+    navigate(-1);
+  }, [navigate]);
+
+  const onControllTextChanged = useCallback((name: string, value: string) => {
+    setForm(form => ({
+      ...form,
+      [name]: value,
+    }));
+  }, []);
+
+  const onNextStep = () => {
+    dispatch(saveTransaction(form));
+    navigate(composeExpenseRoute(groupId, CreateRouteString, ExpenseRouteMode.ADD_MAININFO), {state: { proceed: true }});
```

Comment:
Should we exit this form if the error happen?

Reply:
What error could happen?